src/cogent3/app/__init__.py

```diff
@@ -60,43 +62,42 @@ def _make_types(app) -> dict:
 __apps = None
 
 
-def apps(force: bool = False) -> stevedore.ExtensionManager:
+def get_app_manager(force: bool = False) -> stevedore.ExtensionManager:
     """
     Lazy load a stevedore ExtensionManager to collect apps.
 
-    Parameters
-    ----------
-    force : bool, optional
-        If set to True, forces the re-creation of the ExtensionManager,
-        even if it already exists. Default is False.
-
-    Returns
-    -------
-    ExtensionManager
-        The ExtensionManager instance that collects apps.
+    Notes
+    -----
+    If force is set, the app_manager will refresh it's cache of apps.
     """
     global __apps
     if force:
-        importlib.invalidate_caches()
-        importlib.reload(stevedore)
-        stevedore.extension.ExtensionManager.ENTRY_POINT_CACHE.clear()
-    if not __apps or force:
+        __apps = None  # in some cases setting to None does not reset the global __apps, but creating a new ExtensionManager does
+        __apps = stevedore.ExtensionManager(
+            namespace=APP_ENTRY_POINT, invoke_on_load=False
+        )
+        importlib.invalidate_caches()  # reset the cache of entry points
+        cache = __apps.ENTRY_POINT_CACHE
+        # technique for clearing cache from https://github.com/openstack/cloudkitty/blob/e552c3851fa4bb98096228a89613ceb823d45e72/cloudkitty/utils.py#L198
+        if APP_ENTRY_POINT in cache:
+            del cache[APP_ENTRY_POINT]
+        else:
+            cache.clear()
```

Comment:
why do this if `APP_ENTRY_POINT` is not present?

Reply:
it was an openstack suggestion to handle the feature of implementing dynamic plugins in stevedore.  I was getting desperate at this point and I figgured openstack might  know something I don't about how stevedore works.  Turns out stevedore is just not designed to change it's extensions while the interpreter isn't being reloaded.